pkg/apis/migration/v1alpha1/directvolumemigrationprogress_types.go

```diff
@@ -27,20 +27,65 @@ import (
 
 // DirectVolumeMigrationProgressSpec defines the desired state of DirectVolumeMigrationProgress
 type DirectVolumeMigrationProgressSpec struct {
-	ClusterRef *kapi.ObjectReference `json:"clusterRef,omitempty"`
-	PodRef     *kapi.ObjectReference `json:"podRef,omitempty"`
+	ClusterRef   *kapi.ObjectReference `json:"clusterRef,omitempty"`
+	PodRef       *kapi.ObjectReference `json:"podRef,omitempty"`
+	PodSelector  map[string]string     `json:"podSelector,omitempty"`
+	BackOffLimit int                   `json:"backOffLimit,omitempty"`
 }
 
+const (
+	// DVMPDoneLabelKey used to track progress of dvmp operations
+	DVMPDoneLabelKey = "openshift.migration.io/dvmp-done"
+)
+
 // DirectVolumeMigrationProgressStatus defines the observed state of DirectVolumeMigrationProgress
 type DirectVolumeMigrationProgressStatus struct {
-	Conditions                  `json:",omitempty"`
-	PodPhase                    kapi.PodPhase    `json:"phase,omitempty"`
-	ExitCode                    *int32           `json:"exitCode,omitempty"`
-	ContainerElapsedTime        *metav1.Duration `json:"containerElapsedTime,omitempty"`
-	LogMessage                  string           `json:"logMessage,omitempty"`
-	ObservedDigest              string           `json:"observedDigest,omitempty"`
-	LastObservedProgressPercent string           `json:"lastObservedProgressPercent,omitempty"`
-	LastObservedTransferRate    string           `json:"lastObservedTransferRate,omitempty"`
```

Comment:
Will mig-operator be able to gracefully handle removal of this field? I wonder if we want to avoid removing this, am not sure if it will cause problems. Thinking about clusters where DVMPs already exist.

Reply:
there is also a top level reference to this field in the same struct (used to store status of most recent Rsync attempt). Its just that I encapsulated the overall status in another struct to make it easier to keep history

Reply:
Consider the API is rolled back case, I think you need to keep both for $SOME_PERIOD_OF_TIME

Reply:
@shawn-hurley I have not removed any fields. It's just that these fields will now go into a new struct and that struct is also included "inline" in the Sptatus struct. See line 45, so these new Status fields are compatible with both old and new versions

Reply:
From a JSON perspective it might be fine, what about a user using our APIs from golang?

Reply:
Right, upon upgrade/downgrade, API users attempting to create/update a new DVMPStatus will have to create typed instances of RsyncPodStatus or remove typed status, respectively. Read operations can directly happen through inline field without any change. However, at this point, this CR is only managed by migmigration and controller will take right actions in both old and new version. I wouldn't expect anyone consuming DVM api directly as we don't support running DVM alone.

Reply:
ACK. This field is not going away, just moved to a new struct that is inlined. Users will access the information in the same way. No breaking change from user perspective.